crypto/dilithium/sig_dilithium.h

```diff
@@ -0,0 +1,45 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SIG_DILITHIUM_H
+#define SIG_DILITHIUM_H
+
+#include <stddef.h>
+#include <stdint.h>
+#include <openssl/base.h>
+#include <openssl/evp.h>
+
+// The values below are taken from the |api.h| file in the
+// |crypto/dilithium/pqcrystals_dilithium_ref_common| directory.
+#define DILITHIUM3_PUBLIC_KEY_BYTES 1952
+#define DILITHIUM3_PRIVATE_KEY_BYTES 4000
+#define DILITHIUM3_SIGNATURE_BYTES 3293
+
+// DILITHIUM3_keypair generates a Dilithium3 keypair and assigns a public key to
+// |public_key| and a private key to |secret_key|. The function is a wrapper for
+// the keygen function from the Dilithium source. It returns 0 upon success.
```

Comment:
returns what on error?

Reply:
_it doesn't [look like there's an error path in keygen](https://github.com/aws/aws-lc/blob/integrate-pq/crypto/dilithium/pqcrystals-dilithium_dilithium_common/sign.c#L11-L65)._